src/caip-chain-id.ts

```diff
@@ -0,0 +1,65 @@
+import { is, pattern, string, Struct } from 'superstruct';
+
+import { assert } from './assert';
+
+export type CaipChainId = `${string}:${string}`;
+
+export const CaipChainIdStruct = pattern(
+  string(),
+  /^[-a-z0-9]{3,8}:[-_a-zA-Z0-9]{1,32}$/u,
```

Comment:
i see `u` unicode flag on the Hex superstructs, but i don't understand why it's needed. Is it needed here too?